core/indexer/workQueue.go

```diff
@@ -0,0 +1,92 @@
+package indexer
+
+import (
+	"sync"
+	"time"
+)
+
+const (
+	backOffTime   = time.Second * 10
+	maxBackOff    = time.Minute * 5
+	workCycleTime = time.Second * 2
+)
+
+type workQueue struct {
```

Comment:
I think we should implement this through plain old buffered chans.

Reply:
refactored